src/main/java/net/datafaker/service/WeightedRandomSelector.java

```diff
@@ -0,0 +1,128 @@
+     * @return A randomly selected element based on its weight.
+     * @throws IllegalArgumentException if:
+     *                                  - the list is null or empty,
+     *                                  - any item in the list is null or empty,
+     *                                  - the item does not contain 'weight' or 'value' keys,
+     *                                  - any weight is null, non-positive, NaN or infinite,
+     *                                  - any values in the list are not unique or null,
+     *                                  - the sum of weights exceeds Double.MAX_VALUE.
+     */
+    public <T> T select(List<Map<String, Object>> items) {
+        validateItemsList(items);
+
+        Object[] values = new Object[items.size()];
+        double[] cumulativeWeights = preprocessItems(items, values);
+
+        double randomValue = random.nextDouble() * cumulativeWeights[cumulativeWeights.length - 1];
+        return selectWeightedElement(randomValue, cumulativeWeights, values);
+    }
+
+    private static void validateItemsList(List<Map<String, Object>> items) {
+        if (items == null) {
+            throw new IllegalArgumentException("Input list cannot be null");
+        }
+        if (items.isEmpty()) {
+            throw new IllegalArgumentException("Input list cannot be empty");
+        }
+
+        Set<Object> uniqueValues = new HashSet<>();
+
+        for (var item : items) {
+            validateItem(item);
+            assertUniqueValues(item, uniqueValues);
+        }
+    }
+
+    private static void assertUniqueValues(Map<String, Object> item, Set<Object> values) {
+        Object value = item.get(VALUE_KEY);
+        if (!values.add(value)) {
+            throw new IllegalArgumentException("Duplicate value found: " + value + ". Values must be unique.");
+        }
+    }
+
+    private static void validateItem(Map<String, Object> item) {
+        if (item == null) {
+            throw new IllegalArgumentException("Item cannot be null");
+        }
+        if (item.isEmpty()) {
+            throw new IllegalArgumentException("Item cannot be empty");
+        }
+        if (!item.containsKey(WEIGHT_KEY) || !item.containsKey(VALUE_KEY)) {
+            throw new IllegalArgumentException("Each item must contain 'weight' and 'value' keys");
+        }
+        validateValue(item.get(VALUE_KEY));
+        validateWeight(item.get(WEIGHT_KEY));
+    }
+
+    private static void validateValue(Object valueObj) {
+        if (valueObj == null) {
+            throw new IllegalArgumentException("Value cannot be null");
+        }
+    }
+
+    private static void validateWeight(Object weightObj) {
+        if (!(weightObj instanceof Double weight)) {
+            throw new IllegalArgumentException("Weight must be a non-null Double");
+        }
+        if (weight <= 0 || Double.isNaN(weight) || Double.isInfinite(weight)) {
```

Comment:
out of curiosity: why do we consider `weight = 0` as an error?
IMHO it might be a valid way to turn off this value while the total must be positive

Reply:
My thought was that a weight of 0 implies that the corresponding item cannot be selected. Since WeightedRandomSelector is designed to assign probabilities proportional to weights, including a weight of 0 would lead to unnecessary processing for an element that cannot be selected. But I refactored the code to enable also 0 weight, allowing weights of 0 to be a legitimate way to "turn off" certain elements while still keeping them in the configuration